File: mobile/library/swift/src/stats/Gauge.swift
import Foundation

/// A time series gauge.
@objc
public protocol Gauge: AnyObject {
  /// Set the gauge with the given value.
  func set(value: Int)

  /// Add the given amount to the gauge.
  func add(amount: Int)

  /// Subtract the given amount from the gauge.
  func sub(amount: Int)
}

File: mobile/library/kotlin/io/envoyproxy/envoymobile/stats/Gauge.kt
package io.envoyproxy.envoymobile

/** A time series gauge. */
interface Gauge {

  /** Sets the gauge by the given value. */
  fun set(value: Int)

  /** Sets the gauge by the given value along with tags. */
  fun set(tags: Tags = TagsBuilder().build(), value: Int)

  /** Adds the given amount to the gauge. */
  fun add(amount: Int)

  /** Adds the given amount to the gauge along with tags. */
  fun add(tags: Tags = TagsBuilder().build(), amount: Int)

  /** Subtracts the given amount from the gauge. */
  fun sub(amount: Int)

  /** Subtracts the given amount from the gauge along with tags. */
  fun sub(tags: Tags = TagsBuilder().build(), amount: Int)
}
